python/ray/tests/test_runtime_env_validation.py

```diff
@@ -205,6 +206,23 @@ def test_validate_pip_valid_list(self):
         result = parse_and_validate_pip(PIP_LIST)
         assert result == PIP_LIST
 
+    def test_remove_ray(self):
+        result = parse_and_validate_pip(["pkg1", "ray", "pkg2"])
+        assert result == ["pkg1", "pkg2"]
+
+    def test_remove_ray_env_var(self, monkeypatch):
+        monkeypatch.setenv(RAY_RUNTIME_ENV_ALLOW_RAY_IN_PIP, "1")
+        result = parse_and_validate_pip(["pkg1", "ray", "pkg2"])
+        assert result == ["pkg1", "ray", "pkg2"]
+
+    def test_replace_ray_libraries_with_dependencies(self):
+        result = parse_and_validate_pip(["pkg1", "ray[serve, tune]", "pkg2"])
```

Comment:
is it valid to have a space between `serve, tune`?

Reply:
Oh wow, good catch.  It's valid in a requirements.txt, and it's valid in the shell if you use quotes, but it turns out like we're pasting it into a raw bash command like `pip install ray[serve, tune]`, so it'll fail because of the whitespace.  Let me update the PR to wrap each element of the pip list in quotes before we paste it in the bash command.

Reply:
Oh never mind, it seems seems to just work, the spaces within each argument must somehow be handled automatically in `subprocess.Popen(cmd: List[str], ...)`.  We have an end to end test with `ray[serve, tune]` which passes with this PR.

Reply:
thanks for checking